auparse/auparse_test.go

```diff
@@ -123,41 +142,182 @@ func TestExtractKeyValuePairs(t *testing.T) {
 		},
 		{
 			`x='grep "test" file' y=z`,
-			map[string]*field{
+			fieldMap{
 				"x": {`'grep "test" file'`, `grep "test" file`},
 				"y": newField("z"),
 			},
 		},
 		{
 			`x="grep 'test' file" y=z`,
-			map[string]*field{
+			fieldMap{
 				"x": {`"grep 'test' file"`, `grep 'test' file`},
 				"y": newField("z"),
 			},
 		},
 		{
 			`x="grep \"test\" file" y=z`,
-			map[string]*field{
+			fieldMap{
 				"x": {`"grep \"test\" file"`, `grep \"test\" file`},
 				"y": newField("z"),
 			},
 		},
 		{
 			`x='grep \'test\' file' y=z`,
-			map[string]*field{
+			fieldMap{
 				"x": {`'grep \'test\' file'`, `grep \'test\' file`},
 				"y": newField("z"),
 			},
 		},
 	}
 
 	for _, tc := range tests {
-		out := map[string]*field{}
-		extractKeyValuePairs(tc.in, out)
+		out := extractKeyValuePairs(tc.in)
 		assert.Equal(t, tc.out, out, "failed on: %v", tc.in)
 	}
 }
 
+func BenchmarkExtractKeyValuePairs(b *testing.B) {
+	tests := []struct {
+		name string
+		in   string
+		out  fieldMap
+	}{
+		{
+			"bpf message",
+			`arch=c000003e syscall=321 success=yes exit=0 a0=2 a1=c0059cd6e8 comm="pf-host-agent" exe="/optimyze.cloud/prodfiler/pf-host-agent/pf-host-agent" subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 key="bpf" ARCH=x86_64 SYSCALL=bpf`,
+			fieldMap{
+				"arch":    newField("c000003e"),
+				"syscall": newField("321"),
+				"success": {`yes`, `yes`},
+				"exit":    newField("0"),
+				"a0":      newField("2"),
+				"a1":      {"c0059cd6e8", "c0059cd6e8"},
+				"comm":    {`"pf-host-agent"`, `pf-host-agent`},
+				"exe":     {`"/optimyze.cloud/prodfiler/pf-host-agent/pf-host-agent"`, `/optimyze.cloud/prodfiler/pf-host-agent/pf-host-agent`},
+				"subj":    newField("unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023"),
+				"key":     {`"bpf"`, "bpf"},
+			},
+		},
+		{
+			"short message",
+			`x='grep "test" file' y=z`,
+			fieldMap{
+				"x": {`'grep "test" file'`, `grep "test" file`},
+				"y": newField("z"),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
```

Comment:
These copies are not needed here and below; the closure completes before the next iteration.